internal/cmd/controller/reconciler/cluster_controller.go

```diff
@@ -167,6 +168,8 @@ func (r *ClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		logger.V(1).Error(err, "Reconcile failed final update to cluster status", "status", cluster.Status)
 	}
 
+	metrics.CollectClusterMetrics(cluster)
```

Comment:
I wonder if in addition to these metrics, we want to record the duration of the `Reconcile` call

Record the start time, and record the duration at the end of the function (`defer`) this could be useful to indicate how frequently the cluster can be reconciled, or indicate issues reconciling beyond simple errors (overload cases?).

Reply:
controller-runtime does this out-of-the-box for us, e.g.:

```
controller_runtime_reconcile_time_seconds_bucket{controller="bundle",le="0.005"} 81
controller_runtime_reconcile_time_seconds_bucket{controller="bundle",le="0.01"} 113
controller_runtime_reconcile_time_seconds_bucket{controller="bundle",le="0.025"} 165
controller_runtime_reconcile_time_seconds_bucket{controller="bundle",le="0.05"} 319
controller_runtime_reconcile_time_seconds_bucket{controller="bundle",le="0.1"} 413
controller_runtime_reconcile_time_seconds_bucket{controller="bundle",le="0.15"} 453
controller_runtime_reconcile_time_seconds_bucket{controller="bundle",le="0.2"} 470
controller_runtime_reconcile_time_seconds_bucket{controller="bundle",le="0.25"} 490
controller_runtime_reconcile_time_seconds_bucket{controller="bundle",le="0.3"} 493
controller_runtime_reconcile_time_seconds_bucket{controller="bundle",le="0.35"} 496
controller_runtime_reconcile_time_seconds_bucket{controller="bundle",le="0.4"} 499
controller_runtime_reconcile_time_seconds_bucket{controller="bundle",le="0.45"} 501
controller_runtime_reconcile_time_seconds_bucket{controller="bundle",le="0.5"} 502
controller_runtime_reconcile_time_seconds_bucket{controller="bundle",le="0.6"} 502
controller_runtime_reconcile_time_seconds_bucket{controller="bundle",le="0.7"} 502
controller_runtime_reconcile_time_seconds_bucket{controller="bundle",le="0.8"} 502
controller_runtime_reconcile_time_seconds_bucket{controller="bundle",le="0.9"} 502
controller_runtime_reconcile_time_seconds_bucket{controller="bundle",le="1"} 503
controller_runtime_reconcile_time_seconds_bucket{controller="bundle",le="1.25"} 503
controller_runtime_reconcile_time_seconds_bucket{controller="bundle",le="1.5"} 503
controller_runtime_reconcile_time_seconds_bucket{controller="bundle",le="1.75"} 503
controller_runtime_reconcile_time_seconds_bucket{controller="bundle",le="2"} 503
controller_runtime_reconcile_time_seconds_bucket{controller="bundle",le="2.5"} 503
controller_runtime_reconcile_time_seconds_bucket{controller="bundle",le="3"} 503
controller_runtime_reconcile_time_seconds_bucket{controller="bundle",le="3.5"} 503
controller_runtime_reconcile_time_seconds_bucket{controller="bundle",le="4"} 503
controller_runtime_reconcile_time_seconds_bucket{controller="bundle",le="4.5"} 503
controller_runtime_reconcile_time_seconds_bucket{controller="bundle",le="5"} 503
controller_runtime_reconcile_time_seconds_bucket{controller="bundle",le="6"} 503
controller_runtime_reconcile_time_seconds_bucket{controller="bundle",le="7"} 503
controller_runtime_reconcile_time_seconds_bucket{controller="bundle",le="8"} 503
controller_runtime_reconcile_time_seconds_bucket{controller="bundle",le="9"} 503
controller_runtime_reconcile_time_seconds_bucket{controller="bundle",le="10"} 503
controller_runtime_reconcile_time_seconds_bucket{controller="bundle",le="15"} 503
controller_runtime_reconcile_time_seconds_bucket{controller="bundle",le="20"} 503
controller_runtime_reconcile_time_seconds_bucket{controller="bundle",le="25"} 503
controller_runtime_reconcile_time_seconds_bucket{controller="bundle",le="30"} 503
controller_runtime_reconcile_time_seconds_bucket{controller="bundle",le="40"} 503
controller_runtime_reconcile_time_seconds_bucket{controller="bundle",le="50"} 503
controller_runtime_reconcile_time_seconds_bucket{controller="bundle",le="60"} 503
controller_runtime_reconcile_time_seconds_bucket{controller="bundle",le="+Inf"} 503
```